src/libraries/System.Text.Json/tests/Serialization/ContinuationTests.cs

```diff
@@ -10,502 +11,314 @@
-                {
-                    DefaultBufferSize = 128,
-                    IgnoreNullValues = ignoreNullValues,
-                };
+            var chunk = new Chunk(json, bufferSize);
+            var sequence = new ReadOnlySequence<byte>(chunk, 0, chunk.Next, chunk.Next.Memory.Length);
 
-                Outer<TestValueType<NestedValueType>> obj = await JsonSerializer.DeserializeAsync<Outer<TestValueType<NestedValueType>>>(stream, readOptions);
-
-                Assert.Equal(new string('x', paddingLength), obj.S);
-                Assert.Equal("Hello", obj.C.A);
-                Assert.Null(obj.C.B);
-                Assert.Equal(42, obj.C.C);
-                Assert.Null(obj.C.D);
-                Assert.Equal(3.14e17f, obj.C.E);
-                Assert.Null(obj.C.F);
-                Assert.True(obj.C.G);
-                Assert.Null(obj.C.H);
-                Assert.Collection(obj.C.I, v => Assert.Equal(42, v), v => Assert.Equal(17, v));
-                Assert.Null(obj.C.J.A);
-                Assert.Equal(7, obj.C.J.B);
-            }
+            var reader = new Utf8JsonReader(sequence, true, default);
+            var array = (ITestObject[])JsonSerializer.Deserialize(ref reader, type, readOptions);
+
+            Assert.NotNull(array);
+            Assert.Equal(1, array.Length);
+            array[0].Verify();
         }
 
         [Theory]
-        [MemberData(nameof(TestData))]
-        public static async Task ContinuationShouldWorkAtAnyPosition_ClassWithParamCtor_Class(int paddingLength, bool ignoreNullValues)
+        [MemberData(nameof(TestData), /* enumeratePayloadTweaks: */ true)]
+        public static void InvalidJsonShouldFailAtAnyPosition_Sequence(string json, int bufferSize, Type type, bool ignoreNullValues)
         {
-            var stream = new MemoryStream();
+            var readOptions = new JsonSerializerOptions { IgnoreNullValues = ignoreNullValues, };
+
+            var chunk = new Chunk(json, bufferSize);
+            var sequence = new ReadOnlySequence<byte>(chunk, 0, chunk.Next, chunk.Next.Memory.Length);
+
+            Assert.Throws<JsonException>(() =>
             {
-                var obj = new Outer<TestClassWithParamCtor<NestedClassWithParamCtor>>
-                {
-                    S = new string('x', paddingLength),
-                    C = new(null, 42, null, 3.14e+17f, null, true, null, new int[] { 42, 17 })
-                    {
-                        A = "Hello",
-                        J = new(null)
-                        {
-                            B = 7,
-                        },
-                    },
-                };
-                await JsonSerializer.SerializeAsync(stream, obj, new JsonSerializerOptions { Converters = { new OuterConverter<TestClass<NestedClass>>() } });
-            }
+                var reader = new Utf8JsonReader(sequence, true, default);
+                JsonSerializer.Deserialize(ref reader, type, readOptions);
```

Comment:
Is there a simple way to also verify that the line number/byte position in line for the error, from the exception is set to the correct/expected value? (same question about the other exception test).

Note: Don't compare the exception message, verbatim, since that can change.

Reply:
+1

Reply:
Almost done with this.